src/interface/variable.hpp

```diff
@@ -258,6 +261,33 @@ class EdgeVariable {
   /// return information string
   std::string info();
 
+  // TODO(JMM): should this be 0,1,2?
+  // Should we return the reference? Or something else?
+  KOKKOS_FORCEINLINE_FUNCTION
+  ParArrayND<T> &Get(int i) {
+    assert(1 <= i && i <= 3);
+    if (i == 1) return (data.x1e);
+    if (i == 2)
+      return (data.x2e);
+    else // i == 3
+      return (data.x3e);
+  }
+  template <typename... Args>
+  KOKKOS_FORCEINLINE_FUNCTION T &operator()(int dir, Args... args) const {
+    assert(1 <= dir && dir <= 3);
+    if (dir == 1) return data.x1e(std::forward<Args>(args)...);
+    if (dir == 2)
+      return data.x2e(std::forward<Args>(args)...);
+    else // dir == 3
+      return data.x3e(std::forward<Args>(args)...);
+  }
```

Comment:
I think I agree - we almost always and even preferably interface with variable packs. I wonder if the variable structs entirely could be removed.

Reply:
The nice thing about the variable object is it owns the underlying pararrays as well as fluxes, coarse buffers, and metadata, so I don't know if we want to remove the concept entirely, but I think it could use some cleanup.

Reply:
Yeah, I think the `Variable` object still needs to exist but the `operator()` stuff does not need to. This would just end up being a second place to maintain when the underlying structure of the data in `Variable` gets changed. I would vote to remove it unless there is somewhere it gets used that is hard to switch to packs.